infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -438,6 +430,8 @@ private void finishAllStartupTracesAsFailures(Throwable reason) {
                   serversAndProbesFuture.completeExceptionally(iex);
                 }
                 serversAndProbesFuture.complete(null);
+
+                tracerUtil.finishSpan(tracingSpan);
```

Comment:
Should we also finish span when `ex != null`? This case is checked above

Reply:
Fixed, although this adds some mess and complexity.